sdk/tests/js/tests/tex-input-validation.js

```diff
@@ -328,19 +342,36 @@ debug("Checking TexSubImage2D: a set of inputs that are valid in GL but invalid
 
 testCases = [
   { target: gl.TEXTURE_2D,
-    format: 0x1903, // GL_RED
-    type: gl.UNSIGNED_BYTE,
-    expectedError: gl.INVALID_ENUM },
-  { target: gl.TEXTURE_2D,
-    format: gl.RGBA,
-    type: gl.BYTE,
-    expectedError: gl.INVALID_ENUM },
-  { target: gl.TEXTURE_2D,
     format: gl.RGBA,
     type: gl.UNSIGNED_BYTE,
     expectedError: gl.NO_ERROR }
 ];
 
+if (contextVersion < 2) {
+  testCases = testCases.concat([
+    { target: gl.TEXTURE_2D,
+      format: 0x1903, // GL_RED
+      type: gl.UNSIGNED_BYTE,
+      expectedError: gl.INVALID_ENUM },
+    { target: gl.TEXTURE_2D,
+      format: gl.RGBA,
+      type: gl.BYTE,
+      expectedError: gl.INVALID_ENUM }
+    ]);
+} else {
+  testCases = testCases.concat([
+    { target: gl.TEXTURE_2D,
+      format: 0x1903, // GL_RED
```

Comment:
Can you use gl.RED here? it should be defined in WebGL2. Then I can merge this fix.  Thank you.